### DISCUSSION TITLE
Search, API autocompletion, in table selection when creating relationships

### DISCUSSION THREAD

**Discussion by MrCasquette** [directus/directus#23873] [Feature Requests]
### Summary

It would be nice to have a search feature on table selection when we create a relation between tables.

### Basic Example

![Capture d’écran 2024-10-17 à 16 54 04](https://github.com/user-attachments/assets/5c981fc0-4979-48cd-be33-a572321f54de)

Here's a case where i have many tables (sometimes translated, sometimes not) And it's very difficult for me to find the right table. (I can't imagine another admin)

Maybe we could have the possibility to filter/search by name of table.
Best part would be to find em translated

### Motivation

It's a common use case to have a lot of tables, especially when you create block/section for a page builder.

### Detailed Design

A simple input on the table selection should be good.
An the table should for example disappear as we write in the input. Letting, only those who looks like the name we are looking for.

### Must Have:
- search input

### Should Have:
- table filter alongside

### Could Have:
- Autocompletion
- table translation

### Won't Have:
- it shouldn't interfere with current design and functionality

### Drawbacks

I assume that it should not be a breaking change or a large code difficulty.

### Alternatives

I have no alternatives now, except to looks for my complete table list.

### Adoption Strategy

I think this will not be too difficult to understand as input search are a common behavior in this kind of list.

**Comment by rijkvanzanten** [comment]
Good one! Right now the main typing interaction is already used for creating a new collection on the fly, but maybe an inline search input in that context menu is a good start.

(Also quick tip you can just write the name of the collection in there if you already know the tech name of it. If it already exists it'll change to the primary color as a confirmation that it's recognized as an existing collection)

**Comment by MrCasquette** [comment]
This would be a major time savior for databases with a lot of collections / tables (like CMS with sections builder)
I have more than 50 tables in my current project. If you think that tables are not sorted, not filtered and the translation is not present depending on the place you are, the search for a table is a chore.

It will be really helpfull. 😃
